python/pyspark/sql/dataframe.py

```diff
@@ -935,19 +1090,32 @@ def sample(self, withReplacement=None, fraction=None, seed=None):
         jdf = self._jdf.sample(*args)
         return DataFrame(jdf, self.sql_ctx)
 
-    @since(1.5)
     def sampleBy(self, col, fractions, seed=None):
         """
         Returns a stratified sample without replacement based on the
         fraction given on each stratum.
 
-        :param col: column that defines strata
-        :param fractions:
+        .. versionadded:: 1.5.0
+
+        Parameters
+        ----------
+        col : :class:`Column` or str
+            column that defines strata
+
+            .. versionchanged:: 3.0
+               Added sampling by a column of :class:`Column`
```

Comment:
Is this `versionchanged` put among parameters intentionally?

Reply:
Yeah because it was the change into the specific parameter `col` (SPARK-25381). Looks like at least this is what pandas does.